### PR TITLE
Add support for response compression in echoserver

### PR DESCRIPTION
**Pull request by rramkumar1**
This is nice to have but will also help in debugging #493 

Verified with following request:

```
$ curl -i -H "Accept-Encoding: gzip" localhost:8080

HTTP/1.1 200 OK
Content-Encoding: gzip
Content-Type: application/json
Date: Tue, 16 Apr 2019 21:33:33 GMT
Content-Length: 10

Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.
```

### CODE REVIEW

File: cmd/echo/app/handlers.go [hunk 1]
```diff
@@ -114,11 +117,32 @@ func echo(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	w.Write(processedData)
 	klog.V(3).Infof("echo: %v, %v, %v", time.Now(), r.UserAgent(), r.RemoteAddr)
 }
 
+// process does additional processing on response data if necessary.
+func process(w http.ResponseWriter, r *http.Request, b []byte) ([]byte, error) {
+	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
```
Comment (freehan) [inline cmd/echo/app/handlers.go]: Just curious if there is any chance it will be double compressed?

Comment (rramkumar1) [reply cmd/echo/app/handlers.go]: What do you mean by double compressed?

Comment (freehan) [reply cmd/echo/app/handlers.go]: If the incoming data is already compressed? Will this takes the data and compress it again?

**Pull request merged**